src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarTownUtil.java

```diff
@@ -67,21 +67,23 @@ public static void setTownPvpFlags(Town town, boolean desiredSetting) {
 	 * @param desiredSetting The value to set pvp and explosions to.
 	 */
 	public static void setTownExplosionFlags(Town town, boolean desiredSetting) {
-		//Set it in the town
-		if (town.getPermissions().explosion != desiredSetting)
-			town.getPermissions().explosion = desiredSetting;
-		//Set it in all plots
-		for(TownBlock townBlock: town.getTownBlocks()) {
-			if (townBlock.getPermissions().explosion != desiredSetting)
-				townBlock.getPermissions().explosion = desiredSetting;
-		}
-		//Set it in all plot groups
-		if(town.getPlotGroups() != null) {
-			for (PlotGroup plotGroup : town.getPlotGroups()) {
-				if (plotGroup.getPermissions().explosion != desiredSetting)
-					plotGroup.getPermissions().explosion = desiredSetting;
+		if (SiegeWarSettings.isCannonsIntegrationEnabled()) {
+			//Set it in the town
+			if (town.getPermissions().explosion != desiredSetting)
+				town.getPermissions().explosion = desiredSetting;
+			//Set it in all plots
+			for (TownBlock townBlock : town.getTownBlocks()) {
+				if (townBlock.getPermissions().explosion != desiredSetting)
+					townBlock.getPermissions().explosion = desiredSetting;
+			}
+			//Set it in all plot groups
+			if (town.getPlotGroups() != null) {
+				for (PlotGroup plotGroup : town.getPlotGroups()) {
+					if (plotGroup.getPermissions().explosion != desiredSetting)
+						plotGroup.getPermissions().explosion = desiredSetting;
+				}
 			}
```

Comment:
The plot groups part needs to be removed, since this causes an NPE if a town has one.

Reply:
- It is important for the siege that the explosions are switched to be on/off in plot groups (to prevent explots).
- Will the other code switches (town/plots) effectively result in the effective group permissions being switched?
....if not, what is the pattern I should use to switch it for the plot groups?

Reply:
Changing explosions in townblocks will change these in groups aswell. See #117 and #120.

Reply:
Perfect. Done